pythia/seo/sunspotter.py

```diff
@@ -689,3 +690,63 @@ def plot_observations(self, obsdate: str, mdi_map: Map = None):
 
         plt.legend(handles=[hek_legend])
         plt.show()
+
+    def rotate_to_midnight(self, obsdate: str, fmt='%Y-%m-%d %H:%M:%S'):
+        """
+        Returns the Longitude at midnight, for a given observation time and date.
+        Parameters
+        ----------
+        obsdate : str
+            The observation time and date.
+        fmt : str, optional
+            The format in which obsdate is represented, by default '%Y-%m-%d %H:%M:%S'
+        Returns
+        -------
+        longitude : u.deg
+            longitude of the observation at midnight.
+        Examples
+        --------
+        >>> from pythia.seo import Sunspotter
+        >>> sunspotter = Sunspotter(timesfits="lookup_timefits.csv", properties="lookup_properties.csv")
+        >>> obsdate = '2000-01-01 12:47:02'
+        >>> sunspotter.rotate_to_midnight(obsdate)
+        <Longitude [4.87918286] deg>
+        """
+        rotator = MidnightRotation()
+        properties = self.get_properties_from_obsdate(obsdate)
+        latitude = properties['hcpos_y'].values * u.deg
+        time_to_nearest_midnight = rotator.get_seconds_to_nearest_midnight(obsdate) * u.s
+        return rotator.get_longitude_at_nearest_midnight(time_to_nearest_midnight, latitude)
```

Comment:
It feels weird to use an object as a container of functions

Reply:
Should I just get make the functions without the object?

Reply:
Not sure, leave it for now. We may get a more specific use of the object and then it may be useful to have them there.